File: vue3_apponintments/src/stores/doctor.js
import { defineStore } from "pinia";
import { ref } from 'vue';

export const doctorStore = defineStore('Doctor', () => {
    // 定义描述用户的对象
    const Doctor = ref({
        DoctorId: '',
        Name: '',
        Username: '',
        Department: '',
        Introduction: '',
        Img: '',
        State:'',
    });


    // 定义设置用户信息的方法
    const setDoctor = (newDoctor) => {
        Doctor.value = { ...newDoctor };
    };

    // 定义移除用户信息的方法
    const removeDoctor = () => {
        user.value = {
        DoctorId: '',
        Name: '',
        Username: '',
        Department: '',
        Introduction: '',
        Img: '',
        State: '',
            // 重置其他用户信息字段
        };
    };

    return {
        Doctor,
        setDoctor,
        removeDoctor
    };
},
{
    persist:true
});

File: gin_vue(5)/gin_vue/models/doctor.go
package models

// Doctor 医生表
type Doctor struct {
	DoctorId     int    `json:"DoctorId"`
	Username     string `json:"Username"`
	Password     string `json:"Password"`
	Name         string `json:"Name"`
	DepartmentId int    `json:"DepartmentId"` //科室id
	Department   string `json:"Department"`   //科室名称
	Introduction string `json:"Introduction"` //简历描述
	Price        int    `json:"Price"`        //预约价格
	Img          string `json:"Img"`
	State        string //医生状态
}

func (Doctor) TableName() string {
	return "doctors"
}
